common/beerocks/bcl/source/son/son_wireless_utils.cpp

```diff
@@ -557,6 +559,82 @@ std::vector<uint8_t> wireless_utils::calc_5g_20MHz_subband_channels(
     return channels;
 }
 
+void wireless_utils::get_channel_preferences(
+    beerocks::message::sWifiChannel supported_channels[],
+    std::map<beerocks::message::sPreference, std::list<beerocks::message::sWifiChannel>>
+        &preferences)
+{
+    std::list<uint8_t> oper_classes;
+    wireless_utils::get_supported_operating_classes(supported_channels, oper_classes);
+
+    for (auto oper_class : oper_classes) {
+        const std::set<uint8_t> &oper_class_channels =
+            operating_classes_list.at(oper_class).channels;
+        std::list<beerocks::message::sWifiChannel> radar_affected_channels;
+        for (uint8_t i = 0; i < beerocks::message::SUPPORTED_CHANNELS_LENGTH; i++) {
+            if (oper_class_channels.find(supported_channels[i].channel) !=
+                oper_class_channels.end()) {
+                if (supported_channels[i].radar_affected)
+                    radar_affected_channels.push_back(supported_channels[i]);
+            }
+        }
+        if (!radar_affected_channels.empty()) {
+            auto pref = beerocks::message::sPreference(
+                {oper_class, 0,
+                 wfa_map::cPreferenceOperatingClasses::eReasonCode::
+                     OPERATION_DISALLOWED_DUE_TO_RADAR_DETECTION_ON_A_DFS_CHANNEL});
+
+            preferences[pref] = std::list<beerocks::message::sWifiChannel>(radar_affected_channels);
+        }
+    }
+}
+
+void wireless_utils::get_supported_operating_classes(
+    beerocks::message::sWifiChannel supported_channels[], std::list<uint8_t> &operating_classes)
+{
+    //TODO handle regulatory domain operating classes
+    auto supported_channels_end = supported_channels + beerocks::message::SUPPORTED_CHANNELS_LENGTH;
+
+    for (auto oper_class : operating_classes_list) {
+        for (auto op_class_channel : oper_class.second.channels) {
+            auto it = std::find(supported_channels, supported_channels_end, op_class_channel);
```

Comment:
op_class_channel is a uint8_t and supported_channels is an array of sWifiChannel... You had to add this comparison between sWifiChannel and uint8_t just to support this find operation. I think that's a bit annoying. Therefore, I think it's better to do it differently as described below.

Reply:
Done.